app/components/UI/QRHardware/QRSigningDetails.tsx

```diff
@@ -115,6 +119,22 @@ const QRSigningDetails = ({
 	const [errorMessage, setErrorMessage] = useState('');
 	const [shouldPause, setShouldPause] = useState(false);
 
+	// ios handled camera perfectly in this situation, we just need to check permission with android.
+	const [hasCameraPermission, setCameraPermission] = useState(
+		Platform.OS === 'ios' || bypassAndroidCameraAccessCheck
```

Comment:
nit
```suggestion
		isIos() || bypassAndroidCameraAccessCheck
```

You can import this method from `app/util/device`